src/compiler.ts

```diff
@@ -7941,6 +7941,16 @@ export class Compiler extends DiagnosticEmitter {
       }
       case ElementKind.FUNCTION_PROTOTYPE: {
         let functionPrototype = <FunctionPrototype>target;
+        let typeParameterNodes = functionPrototype.typeParameterNodes;
+
+        if(typeParameterNodes && typeParameterNodes.length != 0) {
```

Comment:
CI is failing because of self compilation failures here, due to [this bug](https://github.com/AssemblyScript/assemblyscript/issues/1136).

Doing `typeParameterNodes !== null` should fix it